### PR TITLE
Updates to new standards introduced in phpBB 3.1.3 and other fixes

### PR DESCRIPTION
**Pull request by VSEphpbb**

### CODE REVIEW

File: controller/controller.php [hunk 1]
```diff
@@ -63,10 +59,10 @@ public function close_announcement()
 		// Check the link hash to protect against CSRF/XSRF attacks
 		if (!check_link_hash($this->request->variable('hash', ''), 'close_boardannouncement') || !$this->config['board_announcements_dismiss'])
 		{
-			return $this->helper->error($this->user->lang('GENERAL_ERROR'), 200);
+			throw new \phpbb\exception\http_exception(500, 'GENERAL_ERROR');
```
Comment (nickvergessen) [inline controller/controller.php]: throw 403, INVALID_TOKEN ?

File: controller/controller.php [hunk 2]
```diff
@@ -88,6 +83,9 @@ public function close_announcement()
 		$redirect = $this->request->variable('redirect', $this->user->data['session_page']);
 		$redirect = reapply_sid($redirect);
 		redirect($redirect);
+
+		// We shouldn't get here, but throw an http exception just in case
+		throw new \phpbb\exception\http_exception(500, 'GENERAL_ERROR');
```
Comment (nickvergessen) [inline controller/controller.php]: give it a suitable name please

File: controller/controller.php (hunk 1 above)
Comment (VSEphpbb) [inline controller/controller.php]: Well, it's not really a form? Would also prefer to rely on existing phpbb lang keys for these.

Comment (nickvergessen) [inline controller/controller.php]: they are from phpbb,
but general error is really not helpful.

FORM_INVALID is still quite generic, but at least it tells the user that his request was the fault and not that something is wrong with the server (what 500 normally means)

Comment (VSEphpbb) [inline controller/controller.php]: The only problem with form invalid is the language suggests a submit action took place. What about 'NOT_AUTHORISED' You are not authorised to access this area.?

File: controller/controller.php (hunk 2 above)
Comment (VSEphpbb) [inline controller/controller.php]: What would a suitable code be ere? 500 in this case, as we shouldn't reach this point due to the redirect() above?

File: controller/controller.php (hunk 1 above)
Comment (nickvergessen) [inline controller/controller.php]: either that, or https://github.com/phpbb/phpbb/blob/040d451dcca9ae54d8f4b7bdd2f231033765a8f2/phpBB/language/en/common.php#L483

File: controller/controller.php (hunk 2 above)
Comment (nickvergessen) [inline controller/controller.php]: Then just remove it?

Comment (VSEphpbb) [inline controller/controller.php]: Removing it actually results in a Symfony error message "The controller must return a response (null given). Did you forget to add a return statement somewhere in your controller?" which should not be displayed to the users, so it's probably better to throw something.

Comment (michaelcullum) [inline controller/controller.php]: This is fine

File: migrations/v10x/m3_initial_module.php [hunk 3]
```diff
@@ -16,6 +16,18 @@
...
 	/**
+	* Assign migration file dependencies for this migration
+	*
+	* @return array Array of migration files
+	* @static
+	* @access public
+	*/
+	static public function depends_on()
+	{
+		return array('\phpbb\boardannouncements\migrations\v10x\m2_initial_data');
```
Comment (michaelcullum) [inline migrations/v10x/m3_initial_module.php]: m1

File: migrations/v10x/m4_announcements_dismiss_config.php [hunk 4]
```diff
@@ -16,6 +16,18 @@
...
 	/**
+	* Assign migration file dependencies for this migration
+	*
+	* @return array Array of migration files
+	* @static
+	* @access public
+	*/
+	static public function depends_on()
+	{
+		return array('\phpbb\boardannouncements\migrations\v10x\m3_initial_module');
```
Comment (michaelcullum) [inline migrations/v10x/m4_announcements_dismiss_config.php]: m1 (or maybe m2 but not strictly).

**Pull request merged**